src/vote_graph.rs

```diff
@@ -119,6 +119,49 @@ impl<H, N, V> VoteGraph<H, N, V> where
 		(self.base.clone(), self.base_number)
 	}
 
+	/// Adjust the base of the graph. The new base must be an ancestor of the
+	/// old base.
+	///
+	/// Provide an ancestry proof from the old base to the new. The proof
+	/// should be in reverse order from the old base's parent.
+	pub fn adjust_base(&mut self, ancestry_proof: &[H]) {
+		let new_hash = match ancestry_proof.last() {
+			None => return, // empty -- nothing to do.
+			Some(h) => h,
+		};
+
+		// not a valid ancestry proof. TODO: error?
+		if ancestry_proof.len() > self.base_number.as_() { return }
```

Comment:
Maybe this should be `>=`?

Reply:
I think we support block `0` so block `1` could have ancestry of `1`.